### ISSUE TITLE
Hardware encode and transcoding enabled result in PlaybackErrorNoCompatibleStream

### ISSUE THREAD

**Issue by Amasuke** [jellyfin/jellyfin#2404]
**Describe the bug**
When enabling hardware encoding and giving the user rights to transcode the stream will always fail to open with an error "PlaybackErrorNoCompatibleStream"
Disabling hardware encoding makes it work again.

**To Reproduce**

1. enable hardware encoding 
2. enable user to transcode audio and video
3. try to play video
5. if no error, set some lower quality to force transcoding
6. See error

**Expected behavior**
Video should play but doesn't.

**Logs**
[ffmpeg-transcode-3b43f4cb-6d62-4da2-b873-6793b9d7173c.txt](https://github.com/jellyfin/jellyfin/files/4201796/ffmpeg-transcode-3b43f4cb-6d62-4da2-b873-6793b9d7173c.txt)
[jellyfin-logs.txt](https://github.com/jellyfin/jellyfin/files/4201797/jellyfin-logs.txt)

**Screenshots**
![firefox_2020-02-13_22-34-42](https://user-images.githubusercontent.com/10835994/74485665-87a7e680-4ebb-11ea-9559-ae8c70b59ffb.png)
![chrome_2020-02-13_23-49-15](https://user-images.githubusercontent.com/10835994/74485667-8a0a4080-4ebb-11ea-94f3-90a7fdc4266a.png)
![firefox_2020-02-13_23-50-07](https://user-images.githubusercontent.com/10835994/74485696-97bfc600-4ebb-11ea-826d-888515e271f1.png)

**System:**
 - OS: Server: Raspbian upgraded to Buster | Client: Windows 10
 - Browser: Chrome, Firefox (also android app)
 - Jellyfin Version: 10.4.3
 - Installed Plugins: none
 - Reverse proxy: apache

**Comment by Artiume** [comment]
here is your ffmpeg cmd, try it with sudo
`sudo /usr/lib/jellyfin-ffmpeg/ffmpeg -i file:"/mount-media/media-drive/Movies/Alien (1979)/Alien (1979) - [Directors Cut 576p x264 10bit en-ger , sub en-ger].mkv" -map_metadata -1 -map_chapters -1 -threads 0 -map 0:0 -map 0:3 -map -0:s -codec:v:0 h264_omx  -b:v 272000 -maxrate 272000 -bufsize 544000 -force_key_frames:0 "expr:gte(t,0+n_forced*3)" -vf "scale=trunc(min(max(iw\,ih*dar)\,426)/2)*2:trunc(ow/dar/2)*2" -copyts -vsync -1 -codec:a:0 libmp3lame -ac 2 -ab 128000  -f hls -max_delay 5000000 -avoid_negative_ts disabled -start_at_zero -hls_time 3 -individual_header_trailer 0 -hls_segment_type mpegts -start_number 0 -hls_segment_filename "/var/lib/jellyfin/transcoding-temp/transcodes/transcodes/transcodes/transcodes/transcodes/transcodes/5e2e022bbe5e09571a346c7d6437d2b3%d.ts" -hls_playlist_type vod -hls_list_size 0 -y "/var/lib/jellyfin/transcoding-temp/transcodes/transcodes/transcodes/transcodes/transcodes/transcodes/5e2e022bbe5e09571a346c7d6437d2b3.m3u8"`

**Comment by Artiume** [comment]
`Stream #0:0: Video: h264 (High 10), yuv420p10le(tv, smpte170m/bt470bg/bt709, progressive), 718x436 [SAR 64:45 DAR 11488:4905]`

So I had wanted to see if it's a permission issue first. I'm now looking at your nightmare of a video lol. It's 10bit and anamorphic. JF does currently have an anamorphic issue https://github.com/jellyfin/jellyfin/issues/2273#issuecomment-583183996 but it shouldn't be the issue. Most people won't notice it. I'm gonna pin a bunch of stuff I'm going to add the docs so read at your own leisure ;) 

Encoding: H264 10Bit
https://en.wikipedia.org/wiki/Advanced_Video_Coding#Profiles

pixel format: yuv420p10le 
This covers the chroma sampling (yuv420)
https://en.wikipedia.org/wiki/YUV#Y%E2%80%B2UV420p_(and_Y%E2%80%B2V12_or_YV12)_to_RGB888_conversion
[SAR 64:45 DAR 11488:4905]
https://forum.videohelp.com/threads/323530-please-explain-SAR-DAR-PAR

(tv: smpte170m/bt470bg/bt709)
smpte170m=NTSC
https://video.stackexchange.com/a/16842
bt470bg
https://forum.videohelp.com/threads/394596-Color-Matrix
bt709
https://en.wikipedia.org/wiki/YUV#HDTV_with_BT.709
progressive
https://www.afterdawn.com/guides/archive/digital_video_fundamentals-frames_frame_rates_page_2.cfm

---

h264 matrix
https://forum.doom9.org/showthread.php?t=168424

What is Chrominance?
https://www.rtings.com/tv/learn/chroma-subsampling

How progressive codecs are built using I, P and B frames.
https://en.wikipedia.org/wiki/Inter_frame

**Comment by Amasuke** [comment]
So I tried that and it says that the conversion failed.

[ffmpeg_command_result.txt](https://github.com/jellyfin/jellyfin/files/4205224/ffmpeg_command_result.txt)

**Comment by Artiume** [comment]
You have an rpi3 correct? If every video is failing then it's your host's libs most likely. Upgrade to nightly, see if that helps. Did you follow the docs and increase video ram?

**Comment by Amasuke** [comment]
Okay, so I did increase the video ram but I didn't upgrade to nightly.
I don't know why but some packages were outdated even tho I upgraded everything when I went to Buster since jellyfin didn't work on jessie. I still have no idea which package was outdated, but after I upgraded a few I can now use HWA and I don't get the error...
It's probably as you said that some lib was not compatible even tho it should have been a fairly new one.

Thanks for the help Artiume.
